assets/wizards/newspack/views/settings/connections/webhooks/endpoint-actions-modals.tsx

```diff
@@ -0,0 +1,424 @@
+/**
+ * Settings Wizard: Connections > Webhooks > Endpoint Actions Modals
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useState, Fragment } from '@wordpress/element';
+import { CheckboxControl as WpCheckboxControl, Icon, TextControl } from '@wordpress/components';
+
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import { useWizardApiFetch } from '../../../../../hooks/use-wizard-api-fetch';
+import { Card, Button, Notice, Modal, Grid } from '../../../../../../components/src';
+import { getDisplayUrl, getEndpointLabel, getRequestStatusIcon, hasEndpointErrors } from './utils';
+
+const ConfirmationModal = ( {
+	disabled,
+	onConfirm,
+	onClose,
+	title,
+	description,
+}: {
+	disabled?: boolean;
+	onConfirm?: () => void;
+	onClose: () => void;
+	title: string;
+	description: string;
+} ) => {
+	return (
+		<Modal title={ title } onRequestClose={ onClose }>
+			<p>{ description }</p>
+			<Card buttonsCard noBorder className="justify-end">
+				<Button variant="secondary" onClick={ onClose } disabled={ disabled }>
+					{ __( 'Cancel', 'newspack-plugin' ) }
+				</Button>
+				<Button variant="primary" onClick={ onConfirm } disabled={ disabled }>
+					{ __( 'Confirm', 'newspack-plugin' ) }
+				</Button>
+			</Card>
+		</Modal>
+	);
+};
+
+/**
+ * Checkbox control props override.
+ *
+ * @param param WP CheckboxControl Component props.
+ * @return      JSX.Element
+ */
+const CheckboxControl: React.FC< WpCheckboxControlPropsOverride< typeof WpCheckboxControl > > = ( {
+	...props
+} ) => {
+	return <WpCheckboxControl { ...props } />;
+};
+
+/**
+ * Endpoints main GET cache key.
+ */
+const NAMESPACE = '/newspack-settings/connections/webhooks';
```

Comment:
non-blocking nitpick: Might be worth moving these constants into a separate constants file so they are easier to track down in the future.

Reply:
Done 👍